src/rust/engine/process_execution/src/remote.rs

```diff
@@ -606,10 +610,20 @@ impl CommandRunner {
     process: Process,
     context: &Context,
   ) -> Result<FallibleProcessResultWithPlatform, String> {
+    const MAX_RETRIES: u32 = 5;
     let start_time = Instant::now();
     let mut current_operation_name: Option<String> = None;
+    let mut num_retries = 0;
 
     loop {
+      // If we are currently retrying a request, then add an exponential backoff.
+      if num_retries > 0 {
+        let multiplier = thread_rng().gen_range(0, 2_u32.pow(num_retries) + 1);
+        let sleep_time = self.retry_interval_duration * multiplier;
+        trace!("retry sleep: {:?}", sleep_time);
```

Comment:
Thoughts on if this would be useful to be debug? A retry feels significant to me, but I'm not sure if this will be extremely common and too noisy at debug.

Reply:
Yeah, it probably would be more useful at `debug` level.